### PR TITLE
Add destination offset options to dragToElement (closes #982)

### LINKED ISSUE #982: Add dragDestionationOffsetX/Y options to the dragToElement function
Need to set destination offset for the `dragToElement` function. 

In my test I want to drag an element to the right side of another element (so, I also need implementation of: https://github.com/DevExpress/testcafe/issues/981)

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**
/cc @AndreyBelym @helen-dikareva @VasilyStrelyaev

### CODE REVIEW

File: src/client/automation/playback/drag/to-element.js [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
+        this.destinationOffsetY = dragToElementOptions.destinationOffsetY;
     }
 
     _getDestination () {
-        var element = this.destinationElement;
-        var offsets = getDefaultAutomationOffsets(this.destinationElement);
+        var element     = this.destinationElement;
+        var elementRect = positionUtils.getElementRectangle(element);
+        var offsets     = getOffsetOptions(element, this.destinationOffsetX, this.destinationOffsetY);
+        var endPoint    = {
```
Comment (AndreyBelym) [inline src/client/automation/playback/drag/to-element.js]: AFAIR there was a style rule to separate multiline decalrations with a one blank line.

File: src/client/automation/playback/drag/to-offset.js [hunk 2]
```diff
@@ -22,19 +22,17 @@ export default class DragToOffsetAutomation extends DragAutomationBase {
...
-        return {
+        endPoint = {
             x: Math.min(Math.max(0, endPoint.x), maxX),
             y: Math.min(Math.max(0, endPoint.y), maxY)
         };
-    }
 
-    _getDestination () {
         var element = document.documentElement;
         var offsets = {
```
Comment (AndreyBelym) [inline src/client/automation/playback/drag/to-offset.js]: Same as above

File: src/client/automation/playback/drag/to-element.js (hunk 1 above)
Comment (AlexanderMoskovkin) [reply src/client/automation/playback/drag/to-element.js]: yep

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

File: ts-defs/index.d.ts [hunk 3]
```diff
@@ -756,6 +756,23 @@ interface TypeActionOptions extends ClickActionOptions {
     paste?: boolean;
 }
 
+interface DragToElementOptions extends MouseActionOptions {
+    /**
+     * Mouse pointer X coordinate that define a point where the dragToElement action is finished.
```
Comment (lexkazakov) [inline ts-defs/index.d.ts]: Mouse pointer X coordinate that defines a point where the dragToElement action is finished.

File: ts-defs/index.d.ts [hunk 4]
```diff
@@ -756,6 +756,23 @@ interface TypeActionOptions extends ClickActionOptions {
...
+     * If an offset is a positive integer, coordinates are calculated relative to the top-left corner of the destination element.
+     * If an offset is a negative integer, they are calculated relative to the bottom-right corner.
+     * The default is the center of the destination element.
+     */
+    destinationOffsetX?: number;
+    /**
+     * Mouse pointer Y coordinate that define a point where the dragToElement is finished started.
```
Comment (lexkazakov) [inline ts-defs/index.d.ts]: Mouse pointer Y coordinate that defines a point where the dragToElement action is finished.

File: ts-defs/index.d.ts [hunk 5]
```diff
@@ -756,6 +756,23 @@ interface TypeActionOptions extends ClickActionOptions {
     paste?: boolean;
 }
 
+interface DragToElementOptions extends MouseActionOptions {
+    /**
+     * Mouse pointer X coordinate that define a point where the dragToElement action is finished.
+     * If an offset is a positive integer, coordinates are calculated relative to the top-left corner of the destination element.
+     * If an offset is a negative integer, they are calculated relative to the bottom-right corner.
+     * The default is the center of the destination element.
```
Comment (lexkazakov) [inline ts-defs/index.d.ts]: By default, the dragToElement action is finished in the center of the destination element.

File: ts-defs/index.d.ts [hunk 6]
```diff
@@ -756,6 +756,23 @@ interface TypeActionOptions extends ClickActionOptions {
...
+     */
+    destinationOffsetX?: number;
+    /**
+     * Mouse pointer Y coordinate that define a point where the dragToElement is finished started.
+     * If an offset is a positive integer, coordinates are calculated relative to the top-left corner of the destination element.
+     * If an offset is a negative integer, they are calculated relative to the bottom-right corner.
+     * The default is the center of the destination element.
```
Comment (lexkazakov) [inline ts-defs/index.d.ts]: By default, the dragToElement action is finished in the center of the destination element.

**Comment by AlexanderMoskovkin** [comment]
FPR /cc @VasilyStrelyaev

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Pull request merged**